diskimg/DiskImg.h

```diff
@@ -1475,7 +1475,9 @@ class DISKIMG_API A2File {
      * allowed) will be lost.
      *
      * The original unmodified filename is availale through GetRawFileName,
-     * though HFS still suffers from embedded nulls.
+     * which can be optionally passed a size_t pointer to get the size
+     * of the raw file name, which is helpful for getting a HFS filename with

```

Comment:
See comments in issue #46 -- I don't think you can have embedded nulls.

Reply:
I was basing the notion of embedded nulls on the GetFileName() implementation comment in DiskImg.h

https://github.com/fadden/ciderpress/blob/7c637c4e6d2e0f7e0a60676bbda1bcd4618e1234/diskimg/DiskImg.h#L1471-L1475

However, if it's not an issue (either from an not-a-problem standpoint or a not-big-enough-of-an-issue-to-worry-about standpoint), then the current code should work.

Reply:
I think the comment may be mistaken.  I'm not sure where I got that from; maybe [wikipedia](https://en.wikipedia.org/wiki/Hierarchical_File_System), which says "discouraged null and nonprints"?  I don't see how you'd have one in HFS given the way the data is stored on disk.  They appear to be forbidden in HFS+ ([ref](https://superuser.com/questions/204287/what-characters-are-forbidden-in-os-x-filenames)).

The wikipedia page also says, "Files were referenced with unique file IDs rather than file names, and file names could be 255 characters long (although the Finder only supported a maximum of 31 characters)." What I've found in the reference material (e.g. FSMakeFSSpec, page 2-166 in Inside Mac: Files) is that it takes a "Str255" argument with the "full or partial pathname".  So the name of a file is limited to 31, and full paths are limited to 255, but wikipedia misrepresents that.

So I think the DiskImg.h comment is in error, and the generic get-raw function will work fine for HFS.

Reply:
Because HFS was implemented in Pascal strings are limited to 255 but null was not a problem.